File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = accumulate(nums.begin(), nums.begin()+k, 0);
        double new_sum = sum;
        for (int i=k; i<nums.size(); i++){
            new_sum = new_sum + nums[i] - nums[i-k];
            sum = max(sum, new_sum);
        }
        double max_ave = sum/k;
        return max_ave;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        summ = sum(nums[0:k])
        new_sum = summ
        for n in range(k, len(nums)):
            new_sum = new_sum + nums[n] - nums[n-k]
            summ = max(summ, new_sum)

        max_ave = summ/k
        return max_ave